presto-main/src/main/java/com/facebook/presto/execution/executor/TaskExecutor.java

```diff
@@ -514,6 +553,27 @@ private synchronized PrioritizedSplitRunner pollNextSplitWorker()
         return null;
     }
 
+    private void interruptRunawaySplits()
+    {
+        for (RunningSplitInfo splitInfo : runningSplitInfos) {
+            Duration duration = Duration.succinctNanos(ticker.read() - splitInfo.getStartTime());
+            if (duration.compareTo(interruptRunawaySplitsTimeout) < 0) {
+                continue;
+            }
+
+            List<StackTraceElement> stack = asList(splitInfo.thread.getStackTrace());
+            String stackString = stack.stream()
+                    .map(Object::toString)
+                    .collect(joining(lineSeparator()));
+            String splitInfoString = splitInfo.getSplitInfo() + lineSeparator() + stackString;
```

Comment:
Move these two lines into the body of `if`. Then, we can also inline `splitInfoString `

Reply:
By inline you mean put it into the log message? I think it's a little clearer being separate, but sure.